xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/webjar/xwiki-filter/plugin.js

```diff
@@ -173,7 +173,15 @@
         event.data.dataValue.filter(new CKEDITOR.htmlParser.filter({
           elements: {
             img: function (element) {
-              delete element.attributes['data-widget'];
+              const dataWidgetAttribute = element.attributes['data-widget'];
+              const dataCkeUploadIdAttribute = element.attributes['data-cke-upload-id'];
+              const isUploadImageWidget = dataWidgetAttribute === 'uploadimage';
+              // Cleanup data-widget attributes on images. Except for 'uploadimage' which is only cleaned up when
+              // the image is not currently being uploaded (i.e., has a data-cke-upload-id attribute)
+              if (!isUploadImageWidget || isUploadImageWidget && dataCkeUploadIdAttribute === undefined)
```

Comment:
Isn't that `isUploadImageWidget` after the `||` redundant? If not, parenthesis should be used to make the operator precedence clear.

Reply:
`isUploadImageWidget` after the or is required, the parenthesis are superfluous but wouldn't hurt (I had them initially and hesitated before removing them).
Let me add them back.

Reply:
But why is it required? If `isUploadImageWidget` is `false`, the first expression is `true` and the rest isn't evaluated. If `isUploadImageWidget` is `true`, `!isUploadImageWidget` is `false` but then in `isUploadImageWidget && dataCkeUploadIdAttribute === undefined`, `isUploadImageWidget` is always `true` and the truth value only depends on `dataCkeUploadIdAttribute === undefined`.